---
title: "Deconvolution of murine lymph node samples"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 3
    vignette: >
      %\VignetteIndexEntry{Deconvolution of murine lymph node samples}
      %\VignetteEngine{knitr::rmarkdown}
      %\VignetteEncoding{UTF-8}
bibliography: references.bib
geometry: margin=3cm
fontsize: 12pt
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE, dpi = 80, fig.width = 8, fig.height = 5.5, 
  base.dir = ".", fig.path = "", fig.align = "center"
)
```

<!-- Reference: @Li2017 -->

In this vignette, we are going to be analyzing a spatial transcriptomics dataset  (10x Visium)  which consists of three murine lymph nodes, two of them after 48-hour stimulation by a _Mycobacterium smegmatis_. It is from DestVI identifies continuums of cell types in spatial transcriptomics data and can be easily downloaded from here: . As reference, we will use the paired single-cell RNA-seq (10x Chromium) that Blabla et al. provided here.

Firstly, let's load and visualize the samples using packages from the Bioconductor's suite: 

```{r}

```


## Loading data

Firstly, we have to create a `SpatialDDLS` object. This class will be the core in which all the steps will be done. We recommend providing both, the spatial and single-cell transcriptomics data in order to filter and select only those genes that are present in both datasets for further processing. 

```{r}
mousedLN.SDDLS <- createSpatialDDLSobject(
  sc.data = sce, 
  sc.cell.ID.column = "CellID", 
  sc.gene.ID.column = "GeneSymbol",
  st.data = st,
  st.cell.ID.column = "CellID",
  st.gene.ID.column = "GeneSymbol",
  sc.min.counts = 10,
  sc.min.cells = 10
)
```


## Simulation of mixed transcriptional profiles

Now, we are going to simulate the cell composition matrices that will serve to simulate mixed transcriptional profiles with known cell proportions. This is done by the `genMixSpotProp` function in which we can control different aspects such as the number of mixed transcriptional profiles that will be generated, the number of cells used for each mixed profile, etc. These parameters must be decided depending on the reference used and the computational resources. For this example, and as standard reference, we will use ` = 10000` and ` = 200`. For more information about these parameters and others, see the Documentation.

Cell type proportions will be generated by three methods: 

* A random sampling of a Dirichlet distribution. Moreover, in order to add sparsity (coger la explicaciÃ³n del Science Advances | Nature Communications) ...
* Pure mixed transcriptional profiles composed of `n.cells` 
* Transcriptional profiles in which a minimum number of missing cell types will be imposed. This is control by ... argument. 

The relative abundance of samples generated by these criteria can be control through ... parameter. Finally,`genMixSpotProp` will automatically divide the reference cells contained in the `single.cell.real` slot into training and test subsets. 

```{r}
sddls <- genMixedCellProp(
  sddls,
  cell.ID.column = "CellID",
  cell.type.column = "broad_cell_types",
  num.sim.spots = 15000,
  n.cells = 200,
  train.freq.cells = 2/3,
  train.freq.spots = 2/3,
  proportion.method = c(0.5, 0.1, 0.4),
  prob.sparity = 0.5, min.zero.prop = 8,
  balanced.type.cells = FALSE,
  verbose = TRUE
)
```


Then, we can call the `` function which will generate the actual mixed transcriptional profiles using the cell composition matrices generated in the previous step. This step may take a while depending on the number of transcriptional profiles to be simulated, so be patient! In addition, the way the mixed profiles will be generated can be decided. We recommend summing up raw counts, and then normalize samples by log-CPM (AddRawCounts), but other methods are available (see Documentation). 

```{r}
sddls <- simMixedProfiles()
```

## Training a fully-connected neural network using mixed transcriptional profiles

Once we have generated a bunch of mixed transcriptional profiles with a known cell composition, we can train a neural network with our train subset to make it predict the cell type proportions present in the test subset. After training, this model will be capable of making cell type proportion predictions uniquely based on the transcriptional profile of new samples, i.e., each spot in a spatial transcriptomics experiment. The architecture of the netwrok is fully customizable and should be changed depending on the number of genes used as input and the complexity of the biological context that is being studied. In this case, we will use a model with 3 hidden layers, each with 200 neurons, and a training process with a total of 40 epochs. 

```{r}
sddls <- sddls %>% trainDeconvModel(
    num.epochs = 40, num.hidden.layers = 3, 
    num.units = c(200, 200, 100), 
  ) 
```

By default, some metrics about the performance of the model will be displayed. In the end, we can calculate more elaborated metrics using the `` function. It will calculate MAE and MSE per cell type, so that it may give us a hint about with what performance the model is working out for each cell type. These metrics can be visualized using different functions: 

```{r}

```


As it can be seen, the overall performance is good, which means that the model is being able to understand thewhat characterized each cell type in order to make accurate predictions. 

## Deconvolution of the spatial transcriptomics dataset

Finally, we can use our recently trained model to deconvolute the signals of each spot! This step is very quick: 


```{r}

```


Now, let's project these predicted proportions in the spatial coordinates:


```{r}

```


In order to see hidden patterns in the coordinates due to using the same color scale, we can use the `` function to plot every cell type independently: 


```{r}

```


Some conclusions and say that these results are very similar to the ones provided by BlaBla et al. 

## Comparing results with general cell markers


```{r}

```






## References
