<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Deconvolution of mouse lymph node samples • SpatialDDLS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Deconvolution of mouse lymph node samples">
<meta property="og:description" content="SpatialDDLS">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SpatialDDLS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href=".././articles/realModelExample-hq.html">Deconvolution of mouse lymph node samples</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/diegommcc/SpatialDDLS" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Deconvolution of mouse lymph node samples</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/diegommcc/SpatialDDLS/vignettes/realModelExample-hq.Rmd" class="external-link"><code>vignettes/realModelExample-hq.Rmd</code></a></small>
      <div class="hidden name"><code>realModelExample-hq.Rmd</code></div>

    </div>

    
    
<!-- Reference: @Li2017 -->
<p>In this vignette, we will analyze a spatial transcriptomics dataset
(10x Visium) comprising three slides from murine lymph nodes, two of
which obtained after a 48-hour infection with <em>Mycobacterium
smegmatis</em> (<span class="citation">Lopez et al. (2022)</span>). As a
reference, we will use the paired single-cell RNA-seq (10x Chromium)
data from the same study. The raw data is publicly available on GEO (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE173778" class="external-link">GSE173778</a>),
but for ease of use, we have made it available through the
<code>SpatialDDLSdata</code> R data package.</p>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>Firstly, let’s load the required packages and data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://diegommcc.github.io/SpatialDDLS/" class="external-link">"SpatialDDLS"</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'SpatialDDLS'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     saveRDS</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: SummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MatrixGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixStats</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span></span>
<span><span class="co">##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span></span>
<span><span class="co">##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span></span>
<span><span class="co">##     union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SpatialExperiment"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">"ggpubr"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## in case it is not installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"SpatialDDLS"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"diegommcc/SpatialDDLSdata"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/diegommcc/SpatialDDLSdata" class="external-link">"SpatialDDLSdata"</a></span><span class="op">)</span></span>
<span><span class="co"># SingleCellExperiment with scRNA-seq</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MouseDLN.SCE</span><span class="op">)</span> </span>
<span><span class="co"># SpatialExperiment with spatial transcriptomics data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span></span></code></pre></div>
<p>For instance, let’s plot the slides contained in the object ny
condition:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html" class="external-link">spatialCoords</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">X0</span>, <span class="va">X1</span>, color <span class="op">=</span> <span class="va">lymph_node</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mouse lymph nodes by condition"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialdata-cond-1.png" width="2187.5" style="display: block; margin: auto;"></p>
<p>With regards to the single-cell RNA-seq data, preprocessing and
visualization using non-linear dimensionality reduction algorithms could
be performed, but such analysis is outside the scope of this
tutorial.</p>
<div class="section level3">
<h3 id="loading-data-into-a-spatialddls-object">Loading data into a <code>SpatialDDLS</code> object<a class="anchor" aria-label="anchor" href="#loading-data-into-a-spatialddls-object"></a>
</h3>
<p>Now, we need to create a <code>SpatialDDLS</code> object, which will
serve as the central core for all subsequent steps. We suggest including
both the spatial and single-cell transcriptomics data to enable
filtering and selection of only those genes that are present in both
data types for further analysis. Furthermore, we recommend filtering
genes based on their expression levels. Please refer to the
documentation to review the implemented strategies
(<code>sc.filt.genes.cells</code> and <code>sc.filt.genes.cluster</code>
parameters).</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSpatialDDLSobject.html">createSpatialDDLSobject</a></span><span class="op">(</span></span>
<span>  sc.data <span class="op">=</span> <span class="va">MouseDLN.SCE</span>, </span>
<span>  sc.cell.ID.column <span class="op">=</span> <span class="st">"CellID"</span>, </span>
<span>  sc.gene.ID.column <span class="op">=</span> <span class="st">"GeneSymbol"</span>,</span>
<span>  sc.cell.type.column <span class="op">=</span> <span class="st">"broad_cell_types"</span>,</span>
<span>  st.data <span class="op">=</span> <span class="va">MouseDLN.ST</span>,</span>
<span>  sc.filt.genes.cells <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sc.min.counts <span class="op">=</span> <span class="fl">1</span>, </span>
<span>  sc.min.cells <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  sc.filt.genes.cluster <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sc.min.mean.counts <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  st.spot.ID.column <span class="op">=</span> <span class="st">"CellID"</span>,</span>
<span>  st.gene.ID.column <span class="op">=</span> <span class="st">"GeneSymbol"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## === 1 SpatialExperiment objects provided</span></span></code></pre>
<pre><code><span><span class="co">##    === Processing spatial transcriptomics data</span></span></code></pre>
<pre><code><span><span class="co">##       - Filtering features:</span></span></code></pre>
<pre><code><span><span class="co">##          - Selected features: 13948</span></span></code></pre>
<pre><code><span><span class="co">##          - Discarded features: 0</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## === Processing single-cell data</span></span></code></pre>
<pre><code><span><span class="co">##       - Removing 16 genes without expression in any cell</span></span></code></pre>
<pre><code><span><span class="co">##       - Filtering features:</span></span></code></pre>
<pre><code><span><span class="co">##          - Selected features: 1142</span></span></code></pre>
<pre><code><span><span class="co">##          - Discarded features: 11696</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Number of shared genes between single-cell and spatial transcriptomics datasets: 1140</span></span></code></pre>
<pre><code><span><span class="co">##     - Original number of genes of single-cell data: 1142</span></span></code></pre>
<pre><code><span><span class="co">##     - Original number of genes of spatial transcriptomics data (object with more genes): 13948</span></span></code></pre>
<p>We can show basic information about the object:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class SpatialDDLS </span></span>
<span><span class="co">## Real single-cell profiles:</span></span>
<span><span class="co">##   1140 features and 14989 cells</span></span>
<span><span class="co">##   rownames: Ankrd33b Txndc5 Spib ... Spib Ubb Serpinb9 Trbc2 </span></span>
<span><span class="co">##   colnames: CTGAATGCAAGACTGG-1-2 GAACGTTTCAAGGAGC-1-2 TTCCGTGTCTTACCGC-1-0 ... TTCCGTGTCTTACCGC-1-0 AAAGGTAGTAGTGGCA-1-2 GCCCGAATCAGCTTGA-1-2 GACCAATGTGGTCTTA-1-2 </span></span>
<span><span class="co">## Spatial experiments:</span></span>
<span><span class="co">##   1 experiments</span></span>
<span><span class="co">##   1140 features and 1092 cells</span></span>
<span><span class="co">##   rownames: R3hdm4 Pcgf5 Rgs1 ... Rgs1 Sh3bgrl3 H2-Ab1 Rps13 </span></span>
<span><span class="co">##   colnames: CGAAACGCAATTCATG-1-0 TACGCAGTTCTTTCCT-1-1 GCCGATTGGCCAAGCT-1-1 ... GCCGATTGGCCAAGCT-1-1 CCAATTACGGGTCGAG-1-1 AATATCGAGGGTTCTC-1-1 ACTGTAGCACTTTGGA-1-0 </span></span>
<span><span class="co">## Project: SpatialDDLS-Proj</span></span></code></pre>
<p>In this case, we are only working with 1 spatial transcriptomics
dataset, but an arbitrary number of <code>SpatialExperiment</code>
objects can be provided.</p>
</div>
<div class="section level3">
<h3 id="simulation-of-mixed-transcriptional-profiles">Simulation of mixed transcriptional profiles<a class="anchor" aria-label="anchor" href="#simulation-of-mixed-transcriptional-profiles"></a>
</h3>
<p>Now, we are going to simulate the cell composition matrices that will
serve to simulate mixed transcriptional profiles with known cell
proportions. This is done by the <code>genMixedCellProp</code> function
in which we can control different aspects such as the number of mixed
transcriptional profiles that will be generated, the number of cells
used for each mixed profile, etc. These parameters must be decided
depending on the reference and the available computational resources.
For this example, and as standard reference, we will use
<code>num.sim.spots = 15000</code> and <code>n.cells = 200</code>. For
more information about these parameters, see the documentation. These
cell type proportions will be generated by three methods:</p>
<ul>
<li>A random sampling of a Dirichlet distribution. Moreover, in order to
make these proportions more sparse, the <code>prob.sparity</code>
parameter controls the probability of having missing cell types in each
simulated spot, as opposed to a mixture of all cell types.</li>
<li>Pure mixed transcriptional profiles composed of <code>n.cells</code>
cells of the same cell type aggregated.</li>
<li>Transcriptional profiles in which a minimum number of missing cell
types will be imposed. This is control by the <code>min.zero.prop</code>
argument.</li>
</ul>
<p>The relative abundance of samples generated by these criteria can be
control by the <code>proportion.method</code> parameter.
Finally,<code>genMixedCellProp</code> will automatically divide the
reference cells contained in the <code>single.cell.real</code> slot into
training and test subsets, and randomly assign <code>n.cells</code>
cells to form every mixed transcriptional profile.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genMixedCellProp.html">genMixedCellProp</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  cell.ID.column <span class="op">=</span> <span class="st">"CellID"</span>,</span>
<span>  cell.type.column <span class="op">=</span> <span class="st">"broad_cell_types"</span>,</span>
<span>  num.sim.spots <span class="op">=</span> <span class="fl">15000</span>,</span>
<span>  n.cells <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  train.freq.cells <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span>
<span>  train.freq.spots <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span>
<span>  proportion.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">0.3</span>, <span class="fl">0.3</span><span class="op">)</span>,</span>
<span>  prob.sparity <span class="op">=</span> <span class="fl">0.5</span>, </span>
<span>  min.zero.prop <span class="op">=</span> <span class="fl">9</span>,</span>
<span>  balanced.type.cells <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === The number of mixed profiles that will be generated is equal to 15000</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Training set cells by type:</span></span></code></pre>
<pre><code><span><span class="co">##     - B cells: 5573</span></span>
<span><span class="co">##     - CD4 T cells: 1362</span></span>
<span><span class="co">##     - CD8 T cells: 2179</span></span>
<span><span class="co">##     - cDC1s: 67</span></span>
<span><span class="co">##     - cDC2s: 58</span></span>
<span><span class="co">##     - GD T cells: 59</span></span>
<span><span class="co">##     - Macrophages: 70</span></span>
<span><span class="co">##     - Migratory DCs: 199</span></span>
<span><span class="co">##     - Monocytes: 53</span></span>
<span><span class="co">##     - NK cells: 62</span></span>
<span><span class="co">##     - pDCs: 52</span></span>
<span><span class="co">##     - Tregs: 260</span></span></code></pre>
<pre><code><span><span class="co">## === Test set cells by type:</span></span></code></pre>
<pre><code><span><span class="co">##     - B cells: 2786</span></span>
<span><span class="co">##     - CD4 T cells: 681</span></span>
<span><span class="co">##     - CD8 T cells: 1089</span></span>
<span><span class="co">##     - cDC1s: 33</span></span>
<span><span class="co">##     - cDC2s: 29</span></span>
<span><span class="co">##     - GD T cells: 29</span></span>
<span><span class="co">##     - Macrophages: 35</span></span>
<span><span class="co">##     - Migratory DCs: 100</span></span>
<span><span class="co">##     - Monocytes: 26</span></span>
<span><span class="co">##     - NK cells: 31</span></span>
<span><span class="co">##     - pDCs: 26</span></span>
<span><span class="co">##     - Tregs: 130</span></span></code></pre>
<pre><code><span><span class="co">## === Probability matrix for training data:</span></span></code></pre>
<pre><code><span><span class="co">##     - Mixed spots: 10000</span></span>
<span><span class="co">##     - Cell types: 12</span></span></code></pre>
<pre><code><span><span class="co">## === Probability matrix for test data:</span></span></code></pre>
<pre><code><span><span class="co">##     - Mixed spots: 5000</span></span>
<span><span class="co">##     - Cell types: 12</span></span></code></pre>
<pre><code><span><span class="co">## DONE</span></span></code></pre>
<p>Then, we can call the <code>simMixedProfiles</code> function which
will generate the actual mixed transcriptional profiles using the cell
composition matrices generated in the previous step. This step may take
a while depending on the number of transcriptional profiles to be
simulated, so be patient! In addition, the way the mixed profiles will
be generated can be decided. We recommend summing up raw counts, and
then normalize samples by log-CPM
(<code>mixing.function = "AddRawCount"</code>), but other methods are
available (see Documentation).</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simMixedProfiles.html">simMixedProfiles</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, threads <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## === Setting parallel environment to 3 thread(s)</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Generating train bulk samples:</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Generating test bulk samples:</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## DONE</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="training-a-fully-connected-neural-network-using-mixed-transcriptional-profiles">Training a fully-connected neural network using mixed
transcriptional profiles<a class="anchor" aria-label="anchor" href="#training-a-fully-connected-neural-network-using-mixed-transcriptional-profiles"></a>
</h3>
<p>Having generated a set of mixed transcriptional profiles with known
cell composition, we can then train a neural network using the training
subset to predict cell type proportions in the test subset. Once
trained, the model can make cell type proportion predictions based on
the transcriptional profile of new samples, such as each spot in a
spatial transcriptomics experiment. The architecture of the network is
fully customizable and should be adjusted according to the number of
genes used as input and the complexity of the biological context being
studied. For this study, we will employ a model with three hidden
layers, each consisting of 200 neurons, and a training process involving
40 epochs.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/trainDeconvModel.html">trainDeconvModel</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  num.epochs <span class="op">=</span> <span class="fl">40</span>,</span>
<span>  num.hidden.layers <span class="op">=</span> <span class="fl">3</span>, </span>
<span>  num.units <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">200</span>, <span class="fl">200</span><span class="op">)</span></span>
<span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## Loaded Tensorflow version 2.5.3</span></span></code></pre>
<pre><code><span><span class="co">## === Training and test from stored data</span></span></code></pre>
<pre><code><span><span class="co">##     Using only simulated mixed samples</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Using only simulated mixed samples</span></span></code></pre>
<pre><code><span><span class="co">## Model: "SpatialDDLS"</span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Layer (type)                        Output Shape                    Param #     </span></span>
<span><span class="co">## ================================================================================</span></span>
<span><span class="co">## Dense1 (Dense)                      (None, 200)                     228200      </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Activation1 (Activation)            (None, 200)                     0           </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Dropout1 (Dropout)                  (None, 200)                     0           </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Dense2 (Dense)                      (None, 200)                     40200       </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Activation2 (Activation)            (None, 200)                     0           </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Dropout2 (Dropout)                  (None, 200)                     0           </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Dense3 (Dense)                      (None, 200)                     40200       </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Activation3 (Activation)            (None, 200)                     0           </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Dropout3 (Dropout)                  (None, 200)                     0           </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## Dense4 (Dense)                      (None, 12)                      2412        </span></span>
<span><span class="co">## ________________________________________________________________________________</span></span>
<span><span class="co">## ActivationSoftmax (Activation)      (None, 12)                      0           </span></span>
<span><span class="co">## ================================================================================</span></span>
<span><span class="co">## Total params: 311,012</span></span>
<span><span class="co">## Trainable params: 311,012</span></span>
<span><span class="co">## Non-trainable params: 0</span></span>
<span><span class="co">## ________________________________________________________________________________</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Training DNN with 10000 samples:</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Evaluating DNN in test data (5000 samples)</span></span></code></pre>
<pre><code><span><span class="co">##    - loss: 0.084</span></span>
<span><span class="co">##    - accuracy: 0.8136</span></span>
<span><span class="co">##    - mean_absolute_error: 0.0176</span></span>
<span><span class="co">##    - categorical_accuracy: 0.8136</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Generating prediction results using test data</span></span></code></pre>
<pre><code><span><span class="co">## DONE</span></span></code></pre>
<p>By default, some metrics about the performance of the model will be
displayed. More advanced metrics can be calculated using the
<code>calculateEvalMetrics</code> function. This function computes mean
absolute error (MAE) and mean squared error (MSE) metrics per cell type,
which can provide insight into the model’s performance for each cell
type. These metrics can be visualized using various functions:</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateEvalMetrics.html">calculateEvalMetrics</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/distErrorPlot.html">distErrorPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  error <span class="op">=</span> <span class="st">"AbsErr"</span>,</span>
<span>  x.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>, </span>
<span>  error.labels <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>  type <span class="op">=</span> <span class="st">"boxplot"</span>,</span>
<span>  size.point <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="abserr-celltype-1.png" width="2187.5" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/barErrorPlot.html">barErrorPlot</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, error <span class="op">=</span> <span class="st">"MAE"</span>, by <span class="op">=</span> <span class="st">"CellType"</span><span class="op">)</span></span></code></pre></div>
<p><img src="abserr-celltype-2-1.png" width="2187.5" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/distErrorPlot.html">distErrorPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  x.by <span class="op">=</span> <span class="st">"pBin"</span>,</span>
<span>  error <span class="op">=</span> <span class="st">"AbsErr"</span>,</span>
<span>  facet.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>, </span>
<span>  error.label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"boxplot"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="abserr-celltype-sep-1.png" width="2400" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/corrExpPredPlot.html">corrExpPredPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  facet.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  corr <span class="op">=</span> <span class="st">"both"</span>, </span>
<span>  size.point <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `geom_smooth()` using formula = 'y ~ x'</span></span></code></pre>
<p><img src="corr-pred-1.png" width="2400" style="display: block; margin: auto;"></p>
<p>As demonstrated, the overall performance is satisfactory, indicating
that the model is capable of comprehending the distinctive features of
each cell type to provide precise predictions regarding the cell type
composition of the samples.</p>
</div>
<div class="section level3">
<h3 id="deconvolution-of-the-spatial-transcriptomics-dataset">Deconvolution of the spatial transcriptomics dataset<a class="anchor" aria-label="anchor" href="#deconvolution-of-the-spatial-transcriptomics-dataset"></a>
</h3>
<p>Finally, we can use our trained model to deconvolute the signals of
each spot! This step is very quick. As we already loaded the spatial
transcriptomics experiment we wish to deconvolute, calling the
<code>deconvSpatialDDLS</code> function is sufficient. However, if
additional spatial experiments are to be included, see
<code>loadSTProfiles</code>.</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deconvSpatialDDLS.html">deconvSpatialDDLS</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    No 'index.st' provided. Deconvoluting all SpatialExperiment objects contained in the `spatial.experiments` slot</span></span></code></pre>
<pre><code><span><span class="co">## === Setting 0 features that are not present in trained model to zero</span></span></code></pre>
<pre><code><span><span class="co">## === Normalizing and scaling data</span></span></code></pre>
<pre><code><span><span class="co">## === Predicting cell type proportions</span></span></code></pre>
<pre><code><span><span class="co">## DONE</span></span></code></pre>
<p>Now, let’s project these predicted proportions in the spatial
coordinates:</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialPropAll.html">plotSpatialPropAll</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="pred-spatial-1.png" width="2400" style="display: block; margin: auto;"></p>
<p>To reveal hidden patterns in the coordinates caused by using the same
color scale, we can utilize the <code>plotSpatialProp</code> function to
plot each cell type independently:</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">mouseDLN.SDDLS</span><span class="op">@</span><span class="va">trained.model</span><span class="op">@</span><span class="va">cell.types</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="../reference/plotSpatialProp.html">plotSpatialProp</a></span><span class="op">(</span></span>
<span>    <span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span>, cell.type <span class="op">=</span> <span class="va">i</span>, colors <span class="op">=</span> <span class="st">"spectral"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="pred-spatial-sep-1.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-2.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-3.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-4.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-5.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-6.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-7.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-8.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-9.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-10.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-11.png" width="2187.5" style="display: block; margin: auto;"><img src="pred-spatial-sep-12.png" width="2187.5" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="comparing-results-with-general-cell-markers">Comparing results with general cell markers<a class="anchor" aria-label="anchor" href="#comparing-results-with-general-cell-markers"></a>
</h3>
<p>Finally, we can check whether there is a collocation between the
predicted cell type proportions and the expression of some classic
markers for each cell type.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">customMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"B cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cd74"</span>, <span class="st">"Cd19"</span>, <span class="st">"Cd79a"</span>, <span class="st">"Cd79b"</span>, <span class="st">"Ly6d"</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD4 T cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cd4"</span>, <span class="st">"Lef1"</span>, <span class="st">"Fyb"</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD8 T cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cd8b1"</span>, <span class="st">"Cd8a"</span>, <span class="st">"Trac"</span><span class="op">)</span>,</span>
<span>  cDC1s <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Xcr1"</span>, <span class="st">"Irf8"</span><span class="op">)</span>,</span>
<span>  cDC2s <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Irf4"</span>, <span class="st">"Cd4"</span><span class="op">)</span>,</span>
<span>  <span class="st">"GD T cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Il7r"</span>, <span class="st">"Id2"</span><span class="op">)</span>,</span>
<span>  Macrophages <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lyz2"</span>, <span class="st">"Lyz1"</span>, <span class="st">"Cd86"</span>, <span class="st">"Ly6c1"</span><span class="op">)</span>,</span>
<span>  <span class="st">"Migratory DCs"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ccl5"</span>, <span class="st">"Anxa3"</span>, <span class="st">"Fscn1"</span><span class="op">)</span>,</span>
<span>  Monocytes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Fcer1g"</span>, <span class="st">"Cst3"</span>, <span class="st">"Lst1"</span>, <span class="st">"Itgam"</span>, <span class="st">"Kit"</span>, <span class="st">"Fcgr3"</span><span class="op">)</span>,</span>
<span>  <span class="st">"NK cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Nkg7"</span>, <span class="st">"Il2rb"</span>, <span class="st">"Gzma"</span><span class="op">)</span>,</span>
<span>  pDCs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Siglech"</span>, <span class="st">"Plac8"</span>, <span class="st">"Ly6c2"</span>, <span class="st">"Vtsb"</span>, <span class="st">"Zeb2"</span>, <span class="st">"Siglech"</span><span class="op">)</span>,</span>
<span>  Tregs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ikzf2"</span>, <span class="st">"Il2ra"</span>, <span class="st">"Foxp3"</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## calculate z-scores</span></span>
<span><span class="va">exprST</span> <span class="op">&lt;-</span> <span class="va">MouseDLN.ST</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">logCPM</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">exprST</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">meanZscoresCustom</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  .x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">customMarkers</span><span class="op">)</span>, </span>
<span>  .f <span class="op">=</span> <span class="op">~</span><span class="op">{</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">[</span><span class="va">customMarkers</span><span class="op">[[</span><span class="va">.x</span><span class="op">]</span><span class="op">]</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">}</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">cbind</span>, <span class="va">.</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">meanZscoresCustom</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">customMarkers</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">color.z.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, name <span class="op">=</span> <span class="st">"RdBu"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">st.coor</span> <span class="op">&lt;-</span> <span class="fu">SpatialExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html" class="external-link">spatialCoords</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/spatial.experiments.html">spatial.experiments</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">st.coor</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Spatial"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">dfPlotLong</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">st.coor</span>, <span class="va">meanZscoresCustom</span><span class="op">)</span><span class="op">)</span>, </span>
<span>  id.vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Spatial 1"</span>, <span class="st">"Spatial 2"</span><span class="op">)</span>, </span>
<span>  variable.name <span class="op">=</span> <span class="st">"CellType"</span>, value.name <span class="op">=</span> <span class="st">"Zscore"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">dfPlotLong</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">[[</span><span class="st">"Spatial 1"</span><span class="op">]</span><span class="op">]</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">[[</span><span class="st">"Spatial 2"</span><span class="op">]</span><span class="op">]</span>, color <span class="op">=</span> <span class="va">Zscore</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span>  <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mean z-score of cell type markers"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="va">color.z.scores</span>, limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.ticks.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">CellType</span><span class="op">)</span></span></code></pre></div>
<p><img src="unnamed-chunk-10-1.png" width="2700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Lopez2022" class="csl-entry">
Lopez, R., B. Li, H. Keren-Shaul, P. Boyeau, M. Kedmi, D. Pilzer, A.
Jelinski, et al. 2022. <span>“<span class="nocase"><span>D</span>est<span>V</span><span>I</span> identifies
continuums of cell types in spatial transcriptomics data</span>.”</span>
<em>Nat Biotechnol</em> 40 (9): 1360–69.
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
