<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SpatialDDLS">
<title>Get started! Deconvolution of mouse lymph node samples • SpatialDDLS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Get started! Deconvolution of mouse lymph node samples">
<meta property="og:description" content="SpatialDDLS">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialDDLS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href=".././articles/realModelExample-hq.html">Get started! Deconvolution of mouse lymph node samples</a>
    <a class="dropdown-item" href=".././articles/hdf5Backend.html">HDF5 files</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Functions</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/diegommcc/SpatialDDLS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get started! Deconvolution of mouse lymph node samples</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/diegommcc/SpatialDDLS/vignettes/realModelExample-hq.Rmd" class="external-link"><code>vignettes/realModelExample-hq.Rmd</code></a></small>
      <div class="d-none name"><code>realModelExample-hq.Rmd</code></div>
    </div>

    
    
<!-- Reference: @Li2017 -->
<p>In this vignette, we will analyze a spatial transcriptomics dataset
(10x Visium) comprising three slides from murine lymph nodes, two of
which obtained after a 48-hour infection with <em>Mycobacterium
smegmatis</em> (<span class="citation">Lopez et al. (2022)</span>). As a
reference, we will use the paired single-cell RNA-seq (10x Chromium)
data from the same study. The raw data is publicly available on GEO (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE173778" class="external-link">GSE173778</a>),
but for ease of use, we have made it available through the
<code>SpatialDDLSdata</code> R data package.</p>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>Firstly, let’s load the required packages and data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://diegommcc.github.io/SpatialDDLS/" class="external-link">"SpatialDDLS"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">"SpatialExperiment"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">"ggpubr"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## in case it is not installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"SpatialDDLSdata"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"diegommcc/SpatialDDLSdata"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/diegommcc/SpatialDDLSdata" class="external-link">"SpatialDDLSdata"</a></span><span class="op">)</span></span>
<span><span class="co"># SingleCellExperiment with scRNA-seq</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MouseDLN.SCE</span><span class="op">)</span> </span>
<span><span class="co"># SpatialExperiment with spatial transcriptomics data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span></span></code></pre></div>
<p>Let’s explore the spatial transcriptomics data contained in the
<code>MouseDLN.ST</code> object:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html" class="external-link">spatialCoords</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">X0</span>, <span class="va">X1</span>, color <span class="op">=</span> <span class="va">lymph_node</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mouse lymph nodes by condition"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialdata-cond-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<p>With regard to the single-cell RNA-seq data, preprocessing and
visualization could be performed, but such analysis is outside the scope
of this tutorial.</p>
<div class="section level3">
<h3 id="loading-data-into-a-spatialddls-object">Loading data into a <code>SpatialDDLS</code> object<a class="anchor" aria-label="anchor" href="#loading-data-into-a-spatialddls-object"></a>
</h3>
<p>Now, we need to create a <code>SpatialDDLS</code> object, which will
serve as the central core for all subsequent steps. We suggest including
both the spatial and single-cell transcriptomics data to enable
filtering and selection of only those genes that are present in both
data types for further analyses. Additionally, we recommend filtering
genes based on their expression levels in order to reduce the number of
dimensions and consider only meaningful genes. Please refer to the
documentation to review the implemented strategies (specially
<code>sc.n.genes.per.cluster</code> and <code>sc.min.mean.counts</code>
parameters).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSpatialDDLSobject.html">createSpatialDDLSobject</a></span><span class="op">(</span></span>
<span>  sc.data <span class="op">=</span> <span class="va">MouseDLN.SCE</span>, </span>
<span>  sc.cell.ID.column <span class="op">=</span> <span class="st">"CellID"</span>, </span>
<span>  sc.gene.ID.column <span class="op">=</span> <span class="st">"GeneSymbol"</span>,</span>
<span>  sc.cell.type.column <span class="op">=</span> <span class="st">"broad_cell_types"</span>,</span>
<span>  st.data <span class="op">=</span> <span class="va">MouseDLN.ST</span>,</span>
<span>  sc.min.counts <span class="op">=</span> <span class="fl">1</span>, </span>
<span>  sc.min.cells <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sc.filt.genes.cluster <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  sc.n.genes.per.cluster <span class="op">=</span> <span class="fl">150</span>,</span>
<span>  sc.min.mean.counts <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  st.spot.ID.column <span class="op">=</span> <span class="st">"CellID"</span>,</span>
<span>  st.gene.ID.column <span class="op">=</span> <span class="st">"GeneSymbol"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## === 1 SpatialExperiment objects provided</span></span></code></pre>
<pre><code><span><span class="co">##    === Processing spatial transcriptomics data</span></span></code></pre>
<pre><code><span><span class="co">##       - Filtering features:</span></span></code></pre>
<pre><code><span><span class="co">##          - Selected features: 13948</span></span></code></pre>
<pre><code><span><span class="co">##          - Discarded features: 0</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## === Processing single-cell data</span></span></code></pre>
<pre><code><span><span class="co">##       - Removing 16 genes without expression in any cell</span></span></code></pre>
<pre><code><span><span class="co">## 'as(&lt;dgCMatrix&gt;, "dgTMatrix")' is deprecated.</span></span>
<span><span class="co">## Use 'as(., "TsparseMatrix")' instead.</span></span>
<span><span class="co">## See help("Deprecated") and help("Matrix-deprecated").</span></span></code></pre>
<pre><code><span><span class="co">##       - Filtering features:</span></span></code></pre>
<pre><code><span><span class="co">##          - Selected features: 12350</span></span></code></pre>
<pre><code><span><span class="co">##          - Discarded features: 488</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Number of shared genes between single-cell and spatial transcriptomics datasets: 11592</span></span></code></pre>
<pre><code><span><span class="co">##     - Original # genes in single-cell data: 12350</span></span></code></pre>
<pre><code><span><span class="co">##     - Original # genes in ST data (object with the greatest # genes): 13948</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Number of removed mitochondrial genes: 12</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Number of genes after filtering based on logFC: 1058</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Final number of dimensions for further analyses: 1058</span></span></code></pre>
<p>We can show some basic information about the object:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class SpatialDDLS </span></span>
<span><span class="co">## Real single-cell profiles:</span></span>
<span><span class="co">##   1058 features and 14989 cells</span></span>
<span><span class="co">##   rownames: Dgat2 Neil3 Tcf7 ... Tcf7 Rpl19 Rps13 Gimap4 </span></span>
<span><span class="co">##   colnames: ACCCTTGCAAGTAGTA-1-0 ACACCAAGTATGGGAC-1-1 GGAGCAAGTCGCATTA-1-3 ... GGAGCAAGTCGCATTA-1-3 TGTAGACAGCCACTCG-1-1 ATGATCGGTATCAAGA-1-2 CGCATAAGTCAGTCCG-1-3 </span></span>
<span><span class="co">## Spatial experiments:</span></span>
<span><span class="co">##   1 experiments</span></span>
<span><span class="co">##   1058 features and 1092 spots</span></span>
<span><span class="co">##   rownames: Arhgef10l Osm Clec4a3 ... Clec4a3 Ccl17 Tcrg-C2 Actg2 </span></span>
<span><span class="co">##   colnames: CTGGCGACATAAGTCC-1-1 GGTGAAGTACAGGGAT-1-1 TATTCCACTCAGCTCG-1-1 ... TATTCCACTCAGCTCG-1-1 TGTGTCGCGAGTTGCA-1-0 GAAGTGATTTATCGTG-1-0 AAATCGTGTACCACAA-1-1 </span></span>
<span><span class="co">## Project: SpatialDDLS-Proj</span></span></code></pre>
<p>In this case, we are only working on 1 spatial transcriptomics
dataset, but an arbitrary number of <code>SpatialExperiment</code>
objects can be loaded.</p>
</div>
<div class="section level3">
<h3 id="simulation-of-mixed-transcriptional-profiles">Simulation of mixed transcriptional profiles<a class="anchor" aria-label="anchor" href="#simulation-of-mixed-transcriptional-profiles"></a>
</h3>
<p>Now, we are going to simulate cell composition matrices that will
serve to simulate mixed transcriptional profiles with known cell
proportions. This is done by the <code>genMixedCellProp</code> function
in which we can control different aspects such as the number of mixed
transcriptional profiles to be generated, the number of cells used to
simulate each mixed profile, etc. These parameters must be decided
depending on the single-cell reference and the available computational
resources. For this example, and as standard reference, we will use
<code>num.sim.spots = 10000</code> and <code>n.cells = 50</code>. The
cell type compositions of these mixed profiles will be generated by
three methods:</p>
<ul>
<li>A random sampling of a Dirichlet distribution. Moreover, in order to
make these proportions more sparse, the <code>prob.sparity</code>
parameter controls the probability of having missing cell types in each
simulated spot, as opposed to a mixture of all cell types.</li>
<li>Pure mixed transcriptional profiles composed of <code>n.cells</code>
cells of the same cell type aggregated.</li>
<li>Transcriptional profiles in which a minimum number of missing cell
types will be imposed. This is controlled by the
<code>min.zero.prop</code> argument.</li>
</ul>
<p>The relative abundance of samples generated by these criteria can be
controlled by the <code>proportion.method</code> parameter. Finally, the
<code>genMixedCellProp</code> function will automatically divide the
reference cell profiles contained in the <code>single.cell.real</code>
slot into training and test subsets and randomly assign
<code>n.cells</code> cells to generate every mixed transcriptional
profile.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genMixedCellProp.html">genMixedCellProp</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  cell.ID.column <span class="op">=</span> <span class="st">"CellID"</span>,</span>
<span>  cell.type.column <span class="op">=</span> <span class="st">"broad_cell_types"</span>,</span>
<span>  num.sim.spots <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  n.cells <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  train.freq.cells <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span>
<span>  train.freq.spots <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span>
<span>  proportion.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">0.3</span>, <span class="fl">0.3</span><span class="op">)</span>,</span>
<span>  prob.sparity <span class="op">=</span> <span class="fl">0.8</span>, </span>
<span>  min.zero.prop <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  balanced.type.cells <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === The number of mixed profiles that will be generated is equal to 10000</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Training set cells by type:</span></span></code></pre>
<pre><code><span><span class="co">##     - B cells: 5573</span></span>
<span><span class="co">##     - CD4 T cells: 1362</span></span>
<span><span class="co">##     - CD8 T cells: 2179</span></span>
<span><span class="co">##     - cDC1s: 67</span></span>
<span><span class="co">##     - cDC2s: 58</span></span>
<span><span class="co">##     - GD T cells: 59</span></span>
<span><span class="co">##     - Macrophages: 70</span></span>
<span><span class="co">##     - Migratory DCs: 199</span></span>
<span><span class="co">##     - Monocytes: 53</span></span>
<span><span class="co">##     - NK cells: 62</span></span>
<span><span class="co">##     - pDCs: 52</span></span>
<span><span class="co">##     - Tregs: 260</span></span></code></pre>
<pre><code><span><span class="co">## === Test set cells by type:</span></span></code></pre>
<pre><code><span><span class="co">##     - B cells: 2786</span></span>
<span><span class="co">##     - CD4 T cells: 681</span></span>
<span><span class="co">##     - CD8 T cells: 1089</span></span>
<span><span class="co">##     - cDC1s: 33</span></span>
<span><span class="co">##     - cDC2s: 29</span></span>
<span><span class="co">##     - GD T cells: 29</span></span>
<span><span class="co">##     - Macrophages: 35</span></span>
<span><span class="co">##     - Migratory DCs: 100</span></span>
<span><span class="co">##     - Monocytes: 26</span></span>
<span><span class="co">##     - NK cells: 31</span></span>
<span><span class="co">##     - pDCs: 26</span></span>
<span><span class="co">##     - Tregs: 130</span></span></code></pre>
<pre><code><span><span class="co">## === Probability matrix for training data:</span></span></code></pre>
<pre><code><span><span class="co">##     - Mixed spots: 6667</span></span>
<span><span class="co">##     - Cell types: 12</span></span></code></pre>
<pre><code><span><span class="co">## === Probability matrix for test data:</span></span></code></pre>
<pre><code><span><span class="co">##     - Mixed spots: 3333</span></span>
<span><span class="co">##     - Cell types: 12</span></span></code></pre>
<pre><code><span><span class="co">## DONE</span></span></code></pre>
<p>Then, we can call the <code>simMixedProfiles</code> function, which
will generate the actual mixed transcriptional profiles using the cell
composition matrices generated in the previous step. This step may take
a while depending on the number of transcriptional profiles to be
simulated, so be patient! In addition, users can choose the method by
which the mixed profiles will be generated. We recommend summing up raw
counts, and then normalize samples by log-CPM
(<code>mixing.function = "AddRawCount"</code>), but other methods are
available (see Documentation).</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simMixedProfiles.html">simMixedProfiles</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, threads <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## === Setting parallel environment to 3 thread(s)</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Generating train mixed profiles:</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Generating test mixed profiles:</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## DONE</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="training-a-fully-connected-neural-network-using-mixed-transcriptional-profiles">Training a fully-connected neural network using mixed
transcriptional profiles<a class="anchor" aria-label="anchor" href="#training-a-fully-connected-neural-network-using-mixed-transcriptional-profiles"></a>
</h3>
<p>Having generated a set of mixed transcriptional profiles with known
cell composition, we can then train a neural network using the training
subset and evaluate the model by predicting cell type proportions on the
test subset. Once trained, the model can deconvolute the cellular
composition of new transcriptional profiles, such as each spot in a
spatial transcriptomics experiment. The architecture of the network is
fully customizable, although in our experience, the hyperparameters used
in this example work for most of the cases. Particularly, we will employ
a model with two hidden layers, each consisting of 200 neurons, and a
training process involving 60 epochs.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/trainDeconvModel.html">trainDeconvModel</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  num.epochs <span class="op">=</span> <span class="fl">60</span>,</span>
<span>  num.hidden.layers <span class="op">=</span> <span class="fl">2</span>, </span>
<span>  num.units <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">200</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> </span></code></pre></div>
<pre><code>## 
  1/105 [..............................] - ETA: 12s - loss: 0.0815 - accuracy: 0.7812 - mean_absolute_error: 0.0171 - categorical_accuracy: 0.7812
 79/105 [=====================&gt;........] - ETA: 0s - loss: 0.0729 - accuracy: 0.8896 - mean_absolute_error: 0.0149 - categorical_accuracy: 0.8896 
105/105 [==============================] - 0s 638us/step - loss: 0.0744 - accuracy: 0.8890 - mean_absolute_error: 0.0152 - categorical_accuracy: 0.8890</code></pre>
<p>Some metrics of the training progress can be shown by setting
<code>verbose = TRUE</code>. Anyhow, more advanced metrics can be
calculated using the <code>calculateEvalMetrics</code> function. This
function computes mean absolute error (MAE) and mean squared error (MSE)
metrics per cell type, providing insight into the model’s performance
for each cell type. These metrics can be visualized using various
functions:</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateEvalMetrics.html">calculateEvalMetrics</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/distErrorPlot.html">distErrorPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  error <span class="op">=</span> <span class="st">"AbsErr"</span>,</span>
<span>  x.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>, </span>
<span>  error.labels <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>  type <span class="op">=</span> <span class="st">"boxplot"</span>,</span>
<span>  size.point <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="abserr-celltype-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/distErrorPlot.html">distErrorPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  x.by <span class="op">=</span> <span class="st">"pBin"</span>,</span>
<span>  error <span class="op">=</span> <span class="st">"AbsErr"</span>,</span>
<span>  facet.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>, </span>
<span>  error.label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"boxplot"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="abserr-celltype-sep-1.png" width="1600" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/corrExpPredPlot.html">corrExpPredPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>,</span>
<span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  facet.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  corr <span class="op">=</span> <span class="st">"both"</span>, </span>
<span>  size.point <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `geom_smooth()` using formula = 'y ~ x'</span></span></code></pre>
<p><img src="corr-pred-1.png" width="1600" style="display: block; margin: auto;"></p>
<p>As demonstrated, the overall performance is satisfactory, indicating
that the model is capable of comprehending the distinctive features of
each cell type to provide precise predictions of the cell type
composition of transcriptional profiles.</p>
</div>
<div class="section level3">
<h3 id="deconvolution-of-the-spatial-transcriptomics-dataset">Deconvolution of the spatial transcriptomics dataset<a class="anchor" aria-label="anchor" href="#deconvolution-of-the-spatial-transcriptomics-dataset"></a>
</h3>
<p>Finally, we can use our trained model to deconvolute the signals of
each spot using the `deconvSpatialDDLS’ function. By default, the
function uses the trained model to predict cell proportions of two sets
of transcriptional profiles from the ST datasets:</p>
<ul>
<li>‘Intrinsic’ profiles: these are the actual transcriptional profiles
of every spot in the ST dataset.</li>
<li>‘Extrinsic’ profiles: these are simulated profiles generated from
the surrounding spots of every spot. The concept is to create a mirrored
set of transcriptional profiles that represent the transcriptional
features of the spatial context of each spot.</li>
</ul>
<p>The latter can be used to understand how similar each spot is to its
neighbors. Considering the hypothesis that we can infer the cellular
composition of each spot based on their surroundings given the
correlation between spatial location and cell
composition/transcriptional profile, we can use this information to
spatially contextualize our predictions and improve their accuracy. For
furhter mentions, we refer to this process as regularization. Details
about the methodology are explained in the Documentation and <span class="citation">Mañanes et al. (2023)</span>.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deconvSpatialDDLS.html">deconvSpatialDDLS</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>, k.spots <span class="op">=</span> <span class="fl">6</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    No 'index.st' provided. Deconvoluting all SpatialExperiment objects contained in the `spatial.experiments` slot</span></span></code></pre>
<pre><code><span><span class="co">## === Normalizing data (LogCPM)</span></span></code></pre>
<pre><code><span><span class="co">## === Predicting cell type proportions</span></span></code></pre>
<pre><code>## 
 1/35 [..............................] - ETA: 0s
35/35 [==============================] - 0s 587us/step</code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## === Calculating distances in PCA space</span></span></code></pre>
<pre><code><span><span class="co">##         - Using 133 PCs (variance cutoff: 0.8)</span></span></code></pre>
<pre><code><span><span class="co">## === Calculating alpha factors based on distances</span></span></code></pre>
<pre><code><span><span class="co">## DONE</span></span></code></pre>
<p>Now, let’s project these predicted proportions in the spatial
coordinates:</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialPropAll.html">plotSpatialPropAll</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="pred-spatial-1.png" width="1600" style="display: block; margin: auto;"></p>
<p>To reveal hidden patterns in the coordinates caused by using the same
color scale, we can utilize the <code>plotSpatialProp</code> function to
independently plot each cell type:</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="../reference/trained.model.html">trained.model</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/cell.types.html">cell.types</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/plotSpatialProp.html">plotSpatialProp</a></span><span class="op">(</span></span>
<span>      <span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span>, cell.type <span class="op">=</span> <span class="va">i</span>, </span>
<span>      colors <span class="op">=</span> <span class="st">"blues"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="pred-spatial-sep-1.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-2.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-3.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-4.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-5.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-6.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-7.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-8.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-9.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-10.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-11.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-12.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<p>In addition to the ‘regularized’ cell proportions, we can plot the
predictions calculated for the intrinsic and extrinsic transcriptional
profiles. For instance, let’s plot those predicted from the extrinsic
transcriptional profiles:</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">mouseDLN.SDDLS</span><span class="op">@</span><span class="va">trained.model</span><span class="op">@</span><span class="va">cell.types</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/plotSpatialProp.html">plotSpatialProp</a></span><span class="op">(</span></span>
<span>      <span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span>, cell.type <span class="op">=</span> <span class="va">i</span>, </span>
<span>      colors <span class="op">=</span> <span class="st">"blues"</span>, prediction <span class="op">=</span> <span class="st">"Extrinsic"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="pred-spatial-sep-extrinsic-1.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-2.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-3.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-4.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-5.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-6.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-7.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-8.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-9.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-10.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-11.png" width="1458.33333333333" style="display: block; margin: auto;"><img src="pred-spatial-sep-extrinsic-12.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<p>As one might expect, this is a smoothed version of the final
predictions. It is also possible to visualize distances between
extrinsic and intrinsic transcriptional profiles of each spot to
understand how the regularization step works by using the
<code>plotDistances</code> function:</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDistances.html">plotDistances</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span>, size.point <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="distances-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<p>Those spots with distances less than the mean distance were
regularized according to their nearest neighbor spots.</p>
<!-- We can observe this spots using the following chunk of code:  -->
<!-- ```{r, modified-spots} -->
<!-- st.coor <- SpatialExperiment::spatialCoords( -->
<!--   spatial.experiments(object = mouseDLN.SDDLS, index.st = 1) -->
<!-- )[, 1:2] -->
<!-- colnames(st.coor) <- paste("Spatial", 1:2) -->
<!-- st.pred <- deconv.spots(object = mouseDLN.SDDLS, index.st = 1) -->
<!-- alpha.factors <- st.pred[["Alpha.Factors"]] -->
<!-- dfPlot <- as.data.frame(cbind(st.coor, alpha.factors)) -->
<!-- ggplot( -->
<!--   dfPlot, aes( -->
<!--     x = .data[["Spatial 1"]], y = .data[["Spatial 2"]],  -->
<!--     color = alpha.factors -->
<!--   ) -->
<!-- ) + geom_point(size = 1.5) + scale_color_gradient2() +  -->
<!--   ggtitle("Distances") + SpatialDDLSTheme() + coord_fixed() -->
<!-- ``` -->
</div>
<div class="section level3">
<h3 id="interpreting-the-model">Interpreting the model<a class="anchor" aria-label="anchor" href="#interpreting-the-model"></a>
</h3>
<p>In order to make predictions more transparent,
<code>SpatialDDLS</code> includes an additional module designed to
provide insights into the model’s decision-making process. It relies on
calculating of predicted classes/loss function gradients with respect to
the input variables, a method popularly known as Vanilla Gradient. These
numeric values are computed for each gene and cell type using the pure
mixed transcriptional profiles previously simulated. Therefore, they can
be interpreted as the extent to which each feature is contributing to
the model’s predictions. While these values are initially calculated at
the sample/gene level, they can be aggregated at the cell type level in
order to assess the relevance of different genes for cell type
proportion predictions. These steps are performed through the
<code>interGradientDL</code> function:</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interGradientsDL.html">interGradientsDL</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>, scaling <span class="op">=</span> <span class="st">"standardize"</span>, method <span class="op">=</span> <span class="st">"class"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Importantly, depending on the <code>method</code> parameter, positive
and negative gradient must be differently interpreted:</p>
<ul>
<li>If gradients with respect to the input variables were calculated
using the loss function (<code>method = "loss"</code>), genes with
negative gradients (those that minimize the loss function) will be
positively correlated with the presence of each cell type.</li>
<li>Conversely, if gradients with respect to the input variables were
calculated using classes (<code>method = "class"</code>), genes with
positive gradients (those that make the probability of being a cell type
higher) will be positively associated with each cell type.</li>
</ul>
<p>It is important to note that these markers should not be interpreted
as cell type markers. Rather, they serve as indications to help
interpret the model’s performance. In addition, due to the multivariate
nature of our approach, gradients are surrogates at the feature level
for predictions made considering all input variables collectively, and
thus caution should be exercised in drawing direct conclusions about
specific gene-cell type relationships.</p>
<p>For this example, let’s calculate gradients of the class function
with respect to the input features and show the top 5 genes with the
greatest gradients per cell type:</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top.gradients</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/topGradientsCellType.html">topGradientsCellType</a></span><span class="op">(</span></span>
<span>  <span class="va">mouseDLN.SDDLS</span>, method <span class="op">=</span> <span class="st">"class"</span>, top.n.genes <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span></span>
<span>  <span class="va">top.gradients</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">Positive</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   B cells CD4 T cells CD8 T cells  cDC1s   cDC2s GD T cells   Macrophages</span></span>
<span><span class="co">## 1  Vpreb3       Ly6c1       Cd8b1 Slc1a3 Gm26917     Slc1a3 1300017J02Rik</span></span>
<span><span class="co">## 2  Slc1a3      Igfbp4        Ccr9 Ifi205    Xist        Il4          Cst7</span></span>
<span><span class="co">## 3    Snx2      Slc1a3        Cd8a   Gatm  Slc1a3      Ip6k3       Pla2g2d</span></span>
<span><span class="co">## 4   Man1a       Satb1        Ppic    Lpl Rnf144b      Ramp3          Cbr2</span></span>
<span><span class="co">## 5    Cd72       Trat1        Acp5   Ryr2    Ryr2       Ska3         Timd4</span></span>
<span><span class="co">##   Migratory DCs Monocytes NK cells   pDCs   Tregs</span></span>
<span><span class="co">## 1        Slc1a3    Map4k4   Slc1a3  Ptgs1   Foxp3</span></span>
<span><span class="co">## 2          Xist    Clec4e    Klrc1 Clec2d   Itgb1</span></span>
<span><span class="co">## 3       Gm26917    Slc1a3     Ugcg   Xist   Ctla4</span></span>
<span><span class="co">## 4        Clec2d       B2m     Cd47  Cd200 Izumo1r</span></span>
<span><span class="co">## 5         Pde1c     Pde1c     Jak1  H2-D1   Cxcr3</span></span></code></pre>
<p>As can be seen, among the top 5 genes some canonical markers for
different cell types appear, such as Cd79b for B cells, Cd8a for CD8 T
cells, or Foxp3 for Tregs. These are just the top 5 genes, so
considering a higher number can provide a more comprehensive
understanding of the genes utilizaed by the model.</p>
<p>We also provide the <code>plotHeatmapGradsAgg</code> function for
visualizing N top mean gradients per cell type. This plot highlights
genes with high gradients across different cell types, reflecting the
multivariate nature of neural networks. It is advisable to examine both
the most positive and negative gradients, as these genes significantly
contribute to the network’s predictions.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hh</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotHeatmapGradsAgg.html">plotHeatmapGradsAgg</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, top.n.genes <span class="op">=</span> <span class="fl">4</span>, method <span class="op">=</span> <span class="st">"class"</span><span class="op">)</span></span>
<span><span class="va">hh</span><span class="op">$</span><span class="va">Positive</span></span></code></pre></div>
<p><img src="positive-grad-1.png" width="1200" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hh</span><span class="op">$</span><span class="va">Negative</span></span></code></pre></div>
<p><img src="negative-grad-1.png" width="1200" style="display: block; margin: auto;"></p>
<p>Finally, we can use the <code>plotSpatialGeneExpr</code> to visualize
the spatial distribution of the top N genes per cell tupe in the ST
dataset. Let’s plot genes for some cell types for demonstration
purposes:</p>
<p>the top N genes per cell type and visualize their spatial
distribution in the ST dataset. Let’s plot genes for some cell types
just for demonstration purposes:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top.genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/topGradientsCellType.html">topGradientsCellType</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, top.n.genes <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B cells"</span>, <span class="st">"CD4 T cells"</span>, <span class="st">"CD8 T cells"</span>, <span class="st">"Tregs"</span>, <span class="st">"Monocytes"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">list.plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">top.genes</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Absolute"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">list.plots</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotSpatialGeneExpr.html">plotSpatialGeneExpr</a></span><span class="op">(</span></span>
<span>      <span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span>, gene <span class="op">=</span> <span class="va">j</span>, size.point <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>      title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">i</span>, <span class="st">" - "</span>, <span class="va">j</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="co">## legend removed just for viz</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">list.plots</span>, align <span class="op">=</span> <span class="st">"hv"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="top-genes-spatial-1.png" width="1600" style="display: block; margin: auto;"><img src="top-genes-spatial-2.png" width="1600" style="display: block; margin: auto;"><img src="top-genes-spatial-3.png" width="1600" style="display: block; margin: auto;"><img src="top-genes-spatial-4.png" width="1600" style="display: block; margin: auto;"><img src="top-genes-spatial-5.png" width="1600" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="clustering-analysis">Clustering analysis<a class="anchor" aria-label="anchor" href="#clustering-analysis"></a>
</h3>
<p>The <code>SpatialDDLS</code> R package also includes some functions
to cluster the ST dataset according to the predicted cell composition of
each spot. This functionality enables to dissect the ST datasets into
distinct cellular niches, which can be a relevant information for
further analyses.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouseDLN.SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialPropClustering.html">spatialPropClustering</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span>, k.nn <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    No 'index.st' provided. Deconvoluting all SpatialExperiment objects contained in the `spatial.experiments` slot</span></span></code></pre>
<pre><code><span><span class="co">## === Selected graph-based clustering</span></span></code></pre>
<pre><code><span><span class="co">## === Running clustering for slide 1</span></span></code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialClustering.html">plotSpatialClustering</a></span><span class="op">(</span><span class="va">mouseDLN.SDDLS</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    No 'index.st' provided. Using first ST dataset</span></span></code></pre>
<pre><code><span><span class="co">## === Plotting first clustering configuration Clustering.graph.k.20</span></span></code></pre>
<p><img src="clustering-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="comparing-deconvoluted-cell-proportions-with-colocalization-of-cell-markers">Comparing deconvoluted cell proportions with colocalization of cell
markers<a class="anchor" aria-label="anchor" href="#comparing-deconvoluted-cell-proportions-with-colocalization-of-cell-markers"></a>
</h3>
<p>Finally, we assess whether there is a collocation between the
predicted cell type proportions and the expression of known classic
markers for each cell type. This analysis aims to validate the model’s
predictions by comparing them with well-established cellular markers,
but it does not mean to be a quantitative validation of the model.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">customMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"B cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cd74"</span>, <span class="st">"Cd19"</span>, <span class="st">"Cd79a"</span>, <span class="st">"Cd79b"</span>, <span class="st">"Ly6d"</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD4 T cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cd4"</span>, <span class="st">"Lef1"</span>, <span class="st">"Fyb"</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD8 T cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cd8b1"</span>, <span class="st">"Cd8a"</span>, <span class="st">"Trac"</span><span class="op">)</span>,</span>
<span>  cDC1s <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Xcr1"</span>, <span class="st">"Irf8"</span><span class="op">)</span>,</span>
<span>  cDC2s <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Irf4"</span>, <span class="st">"Cd4"</span><span class="op">)</span>,</span>
<span>  <span class="st">"GD T cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Il7r"</span>, <span class="st">"Id2"</span><span class="op">)</span>,</span>
<span>  Macrophages <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lyz2"</span>, <span class="st">"Lyz1"</span>, <span class="st">"Cd86"</span>, <span class="st">"Ly6c1"</span><span class="op">)</span>,</span>
<span>  <span class="st">"Migratory DCs"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ccl5"</span>, <span class="st">"Anxa3"</span>, <span class="st">"Fscn1"</span><span class="op">)</span>,</span>
<span>  Monocytes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Fcer1g"</span>, <span class="st">"Cst3"</span>, <span class="st">"Lst1"</span>, <span class="st">"Itgam"</span>, <span class="st">"Kit"</span>, <span class="st">"Fcgr3"</span><span class="op">)</span>,</span>
<span>  <span class="st">"NK cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Nkg7"</span>, <span class="st">"Il2rb"</span>, <span class="st">"Gzma"</span><span class="op">)</span>,</span>
<span>  pDCs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Siglech"</span>, <span class="st">"Plac8"</span>, <span class="st">"Ly6c2"</span>, <span class="st">"Vtsb"</span>, <span class="st">"Zeb2"</span>, <span class="st">"Siglech"</span><span class="op">)</span>,</span>
<span>  Tregs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ikzf2"</span>, <span class="st">"Il2ra"</span>, <span class="st">"Foxp3"</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MouseDLN.ST</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## calculate z-scores</span></span>
<span><span class="va">exprST</span> <span class="op">&lt;-</span> <span class="va">MouseDLN.ST</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">logCPM</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">exprST</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">meanZscoresCustom</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  .x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">customMarkers</span><span class="op">)</span>, </span>
<span>  .f <span class="op">=</span> <span class="op">~</span><span class="op">{</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">[</span><span class="va">customMarkers</span><span class="op">[[</span><span class="va">.x</span><span class="op">]</span><span class="op">]</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">}</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">cbind</span>, <span class="va">.</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">meanZscoresCustom</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">customMarkers</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">color.z.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, name <span class="op">=</span> <span class="st">"RdBu"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">st.coor</span> <span class="op">&lt;-</span> <span class="fu">SpatialExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html" class="external-link">spatialCoords</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/spatial.experiments.html">spatial.experiments</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">mouseDLN.SDDLS</span>, index.st <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">st.coor</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Spatial"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">dfPlotLong</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">st.coor</span>, <span class="va">meanZscoresCustom</span><span class="op">)</span><span class="op">)</span>, </span>
<span>  id.vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Spatial 1"</span>, <span class="st">"Spatial 2"</span><span class="op">)</span>, </span>
<span>  variable.name <span class="op">=</span> <span class="st">"CellType"</span>, value.name <span class="op">=</span> <span class="st">"Zscore"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">dfPlotLong</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">[[</span><span class="st">"Spatial 1"</span><span class="op">]</span><span class="op">]</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">[[</span><span class="st">"Spatial 2"</span><span class="op">]</span><span class="op">]</span>, color <span class="op">=</span> <span class="va">Zscore</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span>  <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mean z-score of cell type markers"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="va">color.z.scores</span>, limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.ticks.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">CellType</span><span class="op">)</span> </span></code></pre></div>
<p><img src="zscores-1.png" width="2400" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Lopez2022" class="csl-entry">
Lopez, R., B. Li, H. Keren-Shaul, P. Boyeau, M. Kedmi, D. Pilzer, A.
Jelinski, et al. 2022. <span>“<span class="nocase"><span>D</span>est<span>V</span><span>I</span> identifies
continuums of cell types in spatial transcriptomics data</span>.”</span>
<em>Nat Biotechnol</em> 40 (9): 1360–69.
</div>
<div id="ref-Mananes2023" class="csl-entry">
Mañanes, D., I. Rivero-García, D. Jimenez-Carretero, M. Torres, D.
Sancho, C. Torroja, and F. Sanchez-Cabo. 2023. <span>“<span class="nocase"><span>S</span>patialDDLS: <span>A</span>n <span>R</span>
package to deconvolute spatial transcriptomics data using neural
networks</span>.”</span> <em>bioRxiv</em>.
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
