<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Generate correlation plots between predicted and expected cell type
proportions of test data. Correlation plots can be shown all mixed or either
split by cell type (CellType) or the number of different cell types
present in the spots (nCellTypes)."><title>Generate correlation plots between predicted and expected cell type
proportions of test data — corrExpPredPlot • SpatialDDLS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Generate correlation plots between predicted and expected cell type
proportions of test data — corrExpPredPlot"><meta property="og:description" content="Generate correlation plots between predicted and expected cell type
proportions of test data. Correlation plots can be shown all mixed or either
split by cell type (CellType) or the number of different cell types
present in the spots (nCellTypes)."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialDDLS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href=".././articles/realModelExample-hq.html">Get started! Deconvolution of mouse lymph node samples</a>
    <a class="dropdown-item" href=".././articles/hdf5Backend.html">HDF5 files</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Functions</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/diegommcc/SpatialDDLS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Generate correlation plots between predicted and expected cell type
proportions of test data</h1>
      <small class="dont-index">Source: <a href="https://github.com/diegommcc/SpatialDDLS/R/evalMetrics.R" class="external-link"><code>R/evalMetrics.R</code></a></small>
      <div class="d-none name"><code>corrExpPredPlot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Generate correlation plots between predicted and expected cell type
proportions of test data. Correlation plots can be shown all mixed or either
split by cell type (<code>CellType</code>) or the number of different cell types
present in the spots (<code>nCellTypes</code>).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">corrExpPredPlot</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">colors</span>,</span>
<span>  facet.by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  corr <span class="op">=</span> <span class="st">"both"</span>,</span>
<span>  filter.sc <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pos.x.label <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  pos.y.label <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  sep.labels <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>  size.point <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  alpha.point <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  ncol <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nrow <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  theme <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>object</dt>
<dd><p><code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object with
<code>trained.model</code> slot containing metrics in the
<code>test.deconv.metrics</code> slot of a <code><a href="DeconvDLModel-class.html">DeconvDLModel</a></code>
object.</p></dd>


<dt>colors</dt>
<dd><p>Vector of colors to be used.</p></dd>


<dt>facet.by</dt>
<dd><p>Show data in different panels. Options are <code>nCellTypes</code>
(number of different cell types) and <code>CellType</code> (cell type)
(<code>NULL</code> by default).</p></dd>


<dt>color.by</dt>
<dd><p>Variable used to color data. Options are <code>nCellTypes</code>
and <code>CellType</code>.</p></dd>


<dt>corr</dt>
<dd><p>Correlation value shown as an annotation on the plot. Available
metrics are Pearson's correlation coefficient (<code>'pearson'</code>) and
concordance correlation coefficient (<code>'ccc'</code>). It can be
<code>'pearson'</code>, <code>'ccc'</code> or <code>'both'</code> (by default).</p></dd>


<dt>filter.sc</dt>
<dd><p>Boolean indicating whether single-cell profiles are filtered
out and only mixed transcriptional profile errors are shown (<code>TRUE</code> by
default).</p></dd>


<dt>pos.x.label</dt>
<dd><p>X-axis position of correlation annotations (0.95 by
default).</p></dd>


<dt>pos.y.label</dt>
<dd><p>Y-axis position of correlation annotations (0.1 by
default).</p></dd>


<dt>sep.labels</dt>
<dd><p>Space separating annotations if <code>corr</code> is equal to
<code>'both'</code> (0.15 by default).</p></dd>


<dt>size.point</dt>
<dd><p>Size of points (0.1 by default).</p></dd>


<dt>alpha.point</dt>
<dd><p>Alpha of points (0.1 by default).</p></dd>


<dt>ncol</dt>
<dd><p>Number of columns if <code>facet.by</code> is other than <code>NULL</code>.</p></dd>


<dt>nrow</dt>
<dd><p>Number of rows if <code>facet.by</code> is different from <code>NULL</code>.</p></dd>


<dt>title</dt>
<dd><p>Title of the plot.</p></dd>


<dt>theme</dt>
<dd><p><span class="pkg">ggplot2</span> theme.</p></dd>


<dt>...</dt>
<dd><p>Additional arguments for the <a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a> function
of <span class="pkg">ggplot2</span> if <code>facet.by</code> is not <code>NULL</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A ggplot object.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="calculateEvalMetrics.html">calculateEvalMetrics</a></code> <code><a href="distErrorPlot.html">distErrorPlot</a></code>
<code><a href="blandAltmanLehPlot.html">blandAltmanLehPlot</a></code> <code><a href="barErrorPlot.html">barErrorPlot</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>      <span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">30</span>, lambda <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">15</span>, ncol <span class="op">=</span> <span class="fl">20</span>,</span></span>
<span class="r-in"><span>      dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Cell_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    Cell_Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"CellType"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">20</span>,</span></span>
<span class="r-in"><span>                       replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  rowData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Gene_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="createSpatialDDLSobject.html">createSpatialDDLSobject</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sc.data <span class="op">=</span> <span class="va">sce</span>,</span></span>
<span class="r-in"><span>  sc.cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  sc.gene.ID.column <span class="op">=</span> <span class="st">"Gene_ID"</span>,</span></span>
<span class="r-in"><span>  sc.filt.genes.cluster <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Spatial transcriptomics data not provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Processing single-cell data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       - Filtering features:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Selected features: 15</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Discarded features: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === No mitochondrial genes were found by using ^mt- as regrex</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Final number of dimensions for further analyses: 15</span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="genMixedCellProp.html">genMixedCellProp</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  cell.type.column <span class="op">=</span> <span class="st">"Cell_Type"</span>,</span></span>
<span class="r-in"><span>  num.sim.spots <span class="op">=</span> <span class="fl">50</span>,</span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === The number of mixed profiles that will be generated is equal to 50</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Training set cells by type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType1: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType2: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType3: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType4: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType5: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType6: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Test set cells by type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType1: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType2: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType3: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType4: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType5: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType6: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Probability matrix for training data:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Mixed spots: 34</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Cell types: 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Probability matrix for test data:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Mixed spots: 16</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Cell types: 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="simMixedProfiles.html">simMixedProfiles</a></span><span class="op">(</span><span class="va">SDDLS</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Setting parallel environment to 1 thread(s)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Generating train mixed profiles:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Generating test mixed profiles:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-in"><span><span class="co"># training of DDLS model</span></span></span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="trainDeconvModel.html">trainDeconvModel</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  batch.size <span class="op">=</span> <span class="fl">15</span>,</span></span>
<span class="r-in"><span>  num.epochs <span class="op">=</span> <span class="fl">5</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Training and test from stored data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Using only simulated mixed samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Using only simulated mixed samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Model: "SpatialDDLS"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Layer (type)                   Output Shape               Param #    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> =====================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dense1 (Dense)                 (None, 200)                3200       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> BatchNormalization1 (BatchNorm (None, 200)                800        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Activation1 (Activation)       (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dropout1 (Dropout)             (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dense2 (Dense)                 (None, 200)                40200      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> BatchNormalization2 (BatchNorm (None, 200)                800        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Activation2 (Activation)       (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dropout2 (Dropout)             (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dense3 (Dense)                 (None, 6)                  1206       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> BatchNormalization3 (BatchNorm (None, 6)                  24         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ActivationSoftmax (Activation) (None, 6)                  0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> =====================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Total params: 46,230</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Trainable params: 45,418</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-trainable params: 812</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Training DNN with 34 samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Evaluating DNN in test data (16 samples)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - loss: 1.4361</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - accuracy: 0.125</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - mean_absolute_error: 0.233</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - categorical_accuracy: 0.125</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Generating prediction results using test data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-in"><span><span class="co"># evaluation using test data</span></span></span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="calculateEvalMetrics.html">calculateEvalMetrics</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">SDDLS</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># correlations by cell type</span></span></span>
<span class="r-in"><span><span class="fu">corrExpPredPlot</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  facet.by <span class="op">=</span> <span class="st">"CellType"</span>,</span></span>
<span class="r-in"><span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>,</span></span>
<span class="r-in"><span>  corr <span class="op">=</span> <span class="st">"both"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> `geom_smooth()` using formula = 'y ~ x'</span>
<span class="r-plt img"><img src="corrExpPredPlot-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># correlations of all samples mixed</span></span></span>
<span class="r-in"><span><span class="fu">corrExpPredPlot</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  facet.by <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  color.by <span class="op">=</span> <span class="st">"CellType"</span>,</span></span>
<span class="r-in"><span>  corr <span class="op">=</span> <span class="st">"ccc"</span>,</span></span>
<span class="r-in"><span>  pos.x.label <span class="op">=</span> <span class="fl">0.2</span>,</span></span>
<span class="r-in"><span>  alpha.point <span class="op">=</span> <span class="fl">0.3</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> `geom_smooth()` using formula = 'y ~ x'</span>
<span class="r-plt img"><img src="corrExpPredPlot-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

