<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Estimate the parameters of the ZINB-WaVE model using a real single-cell
RNA-Seq data set as reference to simulate new single-cell profiles and
increase the signal of underrepresented cell types. This step is only is
needed if the size of the single-cell RNA-seq dataset is too small or there
are underrepresented cell types. After this step, the
simSCProfiles function will use the estimated parameters to
simulate new single-cell profiles. See ?simSCProfiles for more
information."><title>Estimate parameters of the ZINB-WaVE model to simulate new single-cell
RNA-Seq expression profiles — estimateZinbwaveParams • SpatialDDLS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Estimate parameters of the ZINB-WaVE model to simulate new single-cell
RNA-Seq expression profiles — estimateZinbwaveParams"><meta property="og:description" content="Estimate the parameters of the ZINB-WaVE model using a real single-cell
RNA-Seq data set as reference to simulate new single-cell profiles and
increase the signal of underrepresented cell types. This step is only is
needed if the size of the single-cell RNA-seq dataset is too small or there
are underrepresented cell types. After this step, the
simSCProfiles function will use the estimated parameters to
simulate new single-cell profiles. See ?simSCProfiles for more
information."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialDDLS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href=".././articles/realModelExample-hq.html">Get started! Deconvolution of mouse lymph node samples</a>
    <a class="dropdown-item" href=".././articles/hdf5Backend.html">HDF5 files</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Functions</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/diegommcc/SpatialDDLS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Estimate parameters of the ZINB-WaVE model to simulate new single-cell
RNA-Seq expression profiles</h1>
      <small class="dont-index">Source: <a href="https://github.com/diegommcc/SpatialDDLS/R/simSingleCell.R" class="external-link"><code>R/simSingleCell.R</code></a></small>
      <div class="d-none name"><code>estimateZinbwaveParams.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Estimate the parameters of the ZINB-WaVE model using a real single-cell
RNA-Seq data set as reference to simulate new single-cell profiles and
increase the signal of underrepresented cell types. This step is only is
needed if the size of the single-cell RNA-seq dataset is too small or there
are underrepresented cell types. After this step, the
<code><a href="simSCProfiles.html">simSCProfiles</a></code> function will use the estimated parameters to
simulate new single-cell profiles. See <code>?<a href="simSCProfiles.html">simSCProfiles</a></code> for more
information.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">estimateZinbwaveParams</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">cell.type.column</span>,</span>
<span>  <span class="va">cell.ID.column</span>,</span>
<span>  <span class="va">gene.ID.column</span>,</span>
<span>  <span class="va">cell.cov.columns</span>,</span>
<span>  <span class="va">gene.cov.columns</span>,</span>
<span>  subset.cells <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  proportional <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  set.type <span class="op">=</span> <span class="st">"All"</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>object</dt>
<dd><p><code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object with a
<code>single.cell.real</code> slot.</p></dd>


<dt>cell.type.column</dt>
<dd><p>Name or column number corresponding to the cell type
of each cell in cells metadata.</p></dd>


<dt>cell.ID.column</dt>
<dd><p>Name or column number corresponding to the cell names
of expression matrix in cells metadata.</p></dd>


<dt>gene.ID.column</dt>
<dd><p>Name or column number corresponding to the notation
used for features/genes in genes metadata.</p></dd>


<dt>cell.cov.columns</dt>
<dd><p>Name or column number(s) in cells metadata to be used
as covariates during model fitting (if no covariates are used, set to empty
or <code>NULL</code>).</p></dd>


<dt>gene.cov.columns</dt>
<dd><p>Name or column number(s) in genes metadata that will
be used as covariates during model fitting (if no covariates are used, set
to empty or <code>NULL</code>).</p></dd>


<dt>subset.cells</dt>
<dd><p>Number of cells to fit the ZINB-WaVE model. Useful when
the original data set is too large to fit the model. Set a value according
to the original data set and the resources available on your computer. If
<code>NULL</code> (by default), all cells will be used. Must be an integer
greater than or equal to the number of cell types considered and less than
or equal to the total number of cells.</p></dd>


<dt>proportional</dt>
<dd><p>If <code>TRUE</code>, the original cell type proportions in the
subset of cells generated by <code>subset.cells</code> will not be altered as far
as possible. If <code>FALSE</code>, all cell types will have the same number of
cells as far as possible (<code>TRUE</code> by default).</p></dd>


<dt>set.type</dt>
<dd><p>Cell type(s) to evaluate (<code>'All'</code> by default). It is
recommended fitting the model to all cell types rather than using only a
subset of them to capture the total variability present in the original
experiment even if only a subset of cell types is simulated.</p></dd>


<dt>threads</dt>
<dd><p>Number of threads used for estimation (1 by default). To set
up the parallel environment, the <span class="pkg">BiocParallel</span> package must be
installed.</p></dd>


<dt>verbose</dt>
<dd><p>Show informative messages during the execution (<code>TRUE</code> by
default).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object with <code>zinb.params</code></p>


<p>slot containing a <code><a href="ZinbParametersModel-class.html">ZinbParametersModel</a></code> object. This
object contains a slot with the estimated ZINB-WaVE parameters from the
real single-cell RNA-Seq data.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>ZINB-WaVE is a flexible model for zero-inflated count data. This function
carries out the model fit to real single-cell data modeling \(Y_{ij}\) (the
count of feature \(j\) for sample \(i\)) as a random variable following a
zero-inflated negative binomial (ZINB) distribution. The estimated parameters
will be used for the simulation of new single-cell expression profiles by
sampling a negative binomial distribution and inserting dropouts from a
binomial distribution. To do so, <span class="pkg">SpatialDDLS</span> uses the
<code><a href="https://rdrr.io/pkg/zinbwave/man/zinbFit.html" class="external-link">zinbFit</a></code> function from the <span class="pkg">zinbwave</span> package
(Risso et al., 2018). For more details about the model, see Risso et al.,
2018.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Risso, D., Perraudeau, F., Gribkova, S. et al. (2018). A general
and flexible method for signal extraction from single-cell RNA-seq data.
Nat Commun 9, 284. doi: <a href="https://doi.org/10.1038/s41467-017-02554-5" class="external-link">doi:10.1038/s41467-017-02554-5</a>
.</p>
<p>Torroja, C. and Sánchez-Cabo, F. (2019). digitalDLSorter: A Deep Learning
algorithm to quantify immune cell populations based on scRNA-Seq data.
Frontiers in Genetics 10, 978. doi: <a href="https://doi.org/10.3389/fgene.2019.00978" class="external-link">doi:10.3389/fgene.2019.00978</a>
.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="simSCProfiles.html">simSCProfiles</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># reproducibility</span></span></span>
<span class="r-in"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>      <span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">30</span>, lambda <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">15</span>, ncol <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>      dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Cell_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    Cell_Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"CellType"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>                       replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  rowData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Gene_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="createSpatialDDLSobject.html">createSpatialDDLSobject</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sc.data <span class="op">=</span> <span class="va">sce</span>,</span></span>
<span class="r-in"><span>  sc.cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  sc.gene.ID.column <span class="op">=</span> <span class="st">"Gene_ID"</span>,</span></span>
<span class="r-in"><span>  project <span class="op">=</span> <span class="st">"Simul_example"</span>,</span></span>
<span class="r-in"><span>  sc.filt.genes.cluster <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Spatial transcriptomics data not provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Processing single-cell data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       - Filtering features:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Selected features: 15</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Discarded features: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === No mitochondrial genes were found by using ^mt- as regrex</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Final number of dimensions for further analyses: 15</span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu">estimateZinbwaveParams</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  cell.type.column <span class="op">=</span> <span class="st">"Cell_Type"</span>,</span></span>
<span class="r-in"><span>  cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  gene.ID.column <span class="op">=</span> <span class="st">"Gene_ID"</span>,</span></span>
<span class="r-in"><span>  subset.cells <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Setting parallel environment to 1 thread(s)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Estimating parameters for all cell types in the experiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Creating cell model matrix based on Cell_Type columns:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	~Cell_Type</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Number of cells for each cell type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType1: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType2: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Creating gene model matrix without gene covariates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Running estimation process (Start time 05:35:11 PM) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Removing genes without expression in any cell</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt;&gt; Fitting ZINB-WaVE model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Create model:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Initialize parameters:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Optimize parameters:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> penalized log-likelihood = -83.120329137823</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After dispersion optimization = -56.133558863131</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    user  system elapsed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   0.026   0.000   0.026 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After right optimization = -54.6057641481247</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After orthogonalization = -54.6057641481247</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    user  system elapsed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   0.012   0.000   0.012 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After left optimization = -54.6055360468243</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After orthogonalization = -54.6055360468243</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> penalized log-likelihood = -54.6055360468243</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After dispersion optimization = -54.6055360468243</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    user  system elapsed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   0.021   0.000   0.020 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After right optimization = -54.605527995469</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After orthogonalization = -54.605527995469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    user  system elapsed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   0.007   0.000   0.008 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After left optimization = -54.6055197876494</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> After orthogonalization = -54.6055197876494</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> penalized log-likelihood = -54.6055197876494</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Invested time: 1.12</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

