<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate training and test cell type composition matrices — genMixedCellProp • SpatialDDLS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Generate training and test cell type composition matrices — genMixedCellProp"><meta property="og:description" content="Generate training and test cell type composition matrices for the simulation
of mixed transcriptional profiles with known cell composition using
single-cell expression profiles. The resulting
PropCellTypes object will contain all the information
needed to simulate new mixed transcriptional profiles. Note this function
does not simulate the mixed profiles, this task is performed by the
simMixedProfiles or trainDeconvModel functions
(see Documentation)."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SpatialDDLS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href=".././articles/realModelExample-hq.html">Deconvolution of mouse lymph node samples</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/diegommcc/SpatialDDLS" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate training and test cell type composition matrices</h1>
    <small class="dont-index">Source: <a href="https://github.com/diegommcc/SpatialDDLS/R/simMixedSpots.R" class="external-link"><code>R/simMixedSpots.R</code></a></small>
    <div class="hidden name"><code>genMixedCellProp.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generate training and test cell type composition matrices for the simulation
of mixed transcriptional profiles with known cell composition using
single-cell expression profiles. The resulting
<code><a href="PropCellTypes-class.html">PropCellTypes</a></code> object will contain all the information
needed to simulate new mixed transcriptional profiles. Note this function
does not simulate the mixed profiles, this task is performed by the
<code><a href="simMixedProfiles.html">simMixedProfiles</a></code> or <code><a href="trainDeconvModel.html">trainDeconvModel</a></code> functions
(see Documentation).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">genMixedCellProp</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">cell.ID.column</span>,</span>
<span>  <span class="va">cell.type.column</span>,</span>
<span>  <span class="va">num.sim.spots</span>,</span>
<span>  n.cells <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  train.freq.cells <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span>
<span>  train.freq.spots <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span>
<span>  proportion.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span><span class="op">)</span>,</span>
<span>  prob.sparity <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  min.zero.prop <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  balanced.type.cells <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p><code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object with
<code>single.cell.real</code> slot and, optionally, with <code>single.cell.simul</code>
slot.</p></dd>


<dt>cell.ID.column</dt>
<dd><p>Name or column number corresponding to cell names in
cells metadata.</p></dd>


<dt>cell.type.column</dt>
<dd><p>Name or column number corresponding to cell types in
cells metadata.</p></dd>


<dt>num.sim.spots</dt>
<dd><p>Number of mixed profiles to be simulated. It is
recommended to adjust this number according to the number of available
single-cell profiles.</p></dd>


<dt>n.cells</dt>
<dd><p>Specifies the number of cells to be randomly selected and
combined to generate the simulated mixed profiles. By default, it is set to
100. It controls the level of noise present in the simulated data, as it
determines how many single-cell profiles will be combined to produce each
spot.</p></dd>


<dt>train.freq.cells</dt>
<dd><p>Proportion of cells used to simulate training mixed
transcriptional profiles (2/3 by default).</p></dd>


<dt>train.freq.spots</dt>
<dd><p>Proportion of mixed transcriptional profiles to be
used for training, relative to the total number of simulated spots
(<code>num.sim.spots</code>). The default value is 2/3.</p></dd>


<dt>proportion.method</dt>
<dd><p>Vector with three elements that controls the
proportion of simulated proportions generated by each method: random
sampling of a Dirichlet distribution, "pure" spots (1 cell type), and spots
with a specified number of different cell types (determined by
<code>min.zero.prop</code>), respectively. By default, 60% of the simulated
proportions will follow a Dirichlet distribution, 20% will consist of only
one cell type, and 20% will have a specified number of different cell
types.</p></dd>


<dt>prob.sparity</dt>
<dd><p>It only affects the proportions generated by the first
method (Dirichlet distribution). It determines the probability of having
missing cell types in each simulated spot, as opposed to a mixture of all
cell types. A higher value for this parameter will result in more sparse
simulated samples.</p></dd>


<dt>min.zero.prop</dt>
<dd><p>This parameter controls the minimum number of cell types
that will be absent in each simulated spot. If <code>NULL</code> (by default),
this value will be half of the total number of different cell types, but
increasing it will result in more spots composed of fewer cell types. This
helps to create more sparse proportions and cover a wider range of
situations during model training.</p></dd>


<dt>balanced.type.cells</dt>
<dd><p>Boolean indicating whether training and test cells
will be split in a balanced way considering cell types (<code>TRUE</code> by
default).</p></dd>


<dt>verbose</dt>
<dd><p>Show informative messages during the execution (<code>TRUE</code> by
default).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object with <code>prob.cell.types</code></p>


<p>slot containing a <code>list</code> with two <code><a href="PropCellTypes-class.html">PropCellTypes</a></code></p>


<p>objects (training and test). For more information about the structure of
this class, see <code>?<a href="PropCellTypes-class.html">PropCellTypes</a></code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>First, the single-cell profiles are randomly divided into two subsets, with
2/3 of the data for training and 1/3 for testing. The default setting for
this ratio can be changed using the <code>train.freq.cells</code> parameter. Next,
a total of <code>num.sim.spots</code> mixed proportions are simulated using a
Dirichlet distribution. This simulation takes into account the probability of
missing cell types in each spot, which can be adjusted using the
<code>prob.sparity</code> parameter. For each mixed sample, <code>n.cells</code>
single-cell profiles are randomly selected and combined to generate the
simulated mixed sample. In addition to the Dirichlet-based proportions, pure
spots (containing only one cell type) and spots containing a specified number
of different cell types (determined by the <code>min.zero.prop</code> patameter)
are also generated in order to cover situations with only a few cell types
present. The proportion of simulated spots generated by each method can be
controlled using the <code>proportion.method</code> parameter. To visualize the
distribution of cell type proportions generated by each method, the
<code><a href="showProbPlot.html">showProbPlot</a></code> function can be used.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="simMixedProfiles.html">simMixedProfiles</a></code> <code><a href="PropCellTypes-class.html">PropCellTypes</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>      <span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">100</span>, lambda <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">40</span>, ncol <span class="op">=</span> <span class="fl">30</span>,</span></span>
<span class="r-in"><span>      dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">40</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Cell_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    Cell_Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"CellType"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span>,</span></span>
<span class="r-in"><span>                       replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  rowData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Gene_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">40</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="createSpatialDDLSobject.html">createSpatialDDLSobject</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sc.data <span class="op">=</span> <span class="va">sce</span>,</span></span>
<span class="r-in"><span>  sc.cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  sc.gene.ID.column <span class="op">=</span> <span class="st">"Gene_ID"</span>,</span></span>
<span class="r-in"><span>  project <span class="op">=</span> <span class="st">"Simul_example"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Spatial transcriptomics data not provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Processing single-cell data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       - Filtering features:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Selected features: 40</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Discarded features: 0</span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu">genMixedCellProp</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  cell.type.column <span class="op">=</span> <span class="st">"Cell_Type"</span>,</span></span>
<span class="r-in"><span>  num.sim.spots <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === The number of mixed profiles that will be generated is equal to 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Training set cells by type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType1: 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType2: 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType3: 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType4: 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Test set cells by type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType1: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType2: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType3: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType4: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Probability matrix for training data:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Mixed spots: 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Cell types: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Probability matrix for test data:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Mixed spots: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Cell types: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-in"><span>  </span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

