<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Train a deep neural network model using training data from the
SpatialDDLS object. This model will be used to
deconvolute spatial transcriptomics data from the same biological context as
the single-cell RNA-seq data used to train it. In addition, the trained
model is evaluated using test data, and prediction results are obtained to
determine its performance (see ?calculateEvalMetrics)."><title>Train deconvolution model for spatial transcriptomics data — trainDeconvModel • SpatialDDLS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Train deconvolution model for spatial transcriptomics data — trainDeconvModel"><meta property="og:description" content="Train a deep neural network model using training data from the
SpatialDDLS object. This model will be used to
deconvolute spatial transcriptomics data from the same biological context as
the single-cell RNA-seq data used to train it. In addition, the trained
model is evaluated using test data, and prediction results are obtained to
determine its performance (see ?calculateEvalMetrics)."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialDDLS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href=".././articles/realModelExample-hq.html">Get started! Deconvolution of mouse lymph node samples</a>
    <a class="dropdown-item" href=".././articles/hdf5Backend.html">HDF5 files</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Functions</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/diegommcc/SpatialDDLS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Train deconvolution model for spatial transcriptomics data</h1>
      <small class="dont-index">Source: <a href="https://github.com/diegommcc/SpatialDDLS/R/dnnModel.R" class="external-link"><code>R/dnnModel.R</code></a></small>
      <div class="d-none name"><code>trainDeconvModel.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Train a deep neural network model using training data from the
<code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object. This model will be used to
deconvolute spatial transcriptomics data from the same biological context as
the single-cell RNA-seq data used to train it. In addition, the trained
model is evaluated using test data, and prediction results are obtained to
determine its performance (see <code>?<a href="calculateEvalMetrics.html">calculateEvalMetrics</a></code>).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">trainDeconvModel</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  type.data.train <span class="op">=</span> <span class="st">"mixed"</span>,</span>
<span>  type.data.test <span class="op">=</span> <span class="st">"mixed"</span>,</span>
<span>  batch.size <span class="op">=</span> <span class="fl">64</span>,</span>
<span>  num.epochs <span class="op">=</span> <span class="fl">60</span>,</span>
<span>  num.hidden.layers <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  num.units <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">200</span><span class="op">)</span>,</span>
<span>  activation.fun <span class="op">=</span> <span class="st">"relu"</span>,</span>
<span>  dropout.rate <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  loss <span class="op">=</span> <span class="st">"kullback_leibler_divergence"</span>,</span>
<span>  metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"accuracy"</span>, <span class="st">"mean_absolute_error"</span>, <span class="st">"categorical_accuracy"</span><span class="op">)</span>,</span>
<span>  normalize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  scaling <span class="op">=</span> <span class="st">"standardize"</span>,</span>
<span>  norm.batch.layers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  custom.model <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shuffle <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sc.downsampling <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  use.generator <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  on.the.fly <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  agg.function <span class="op">=</span> <span class="st">"AddRawCount"</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  view.metrics.plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>object</dt>
<dd><p><code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object with
<code>single.cell.real</code>/<code>single.cell.simul</code>, <code>prob.cell.types</code>,
and <code>mixed.profiles</code> slots (the last only if <code>on.the.fly =
  FALSE</code>).</p></dd>


<dt>type.data.train</dt>
<dd><p>Type of profiles to be used for training. It can be
<code>'both'</code>, <code>'single-cell'</code> or <code>'mixed'</code> (<code>'mixed'</code> by
default).</p></dd>


<dt>type.data.test</dt>
<dd><p>Type of profiles to be used for evaluation. It can be
<code>'both'</code>, <code>'single-cell'</code> or <code>'mixed'</code> (<code>'mixed'</code> by
default).</p></dd>


<dt>batch.size</dt>
<dd><p>Number of samples per gradient update (64 by default).</p></dd>


<dt>num.epochs</dt>
<dd><p>Number of epochs to train the model (60 by default).</p></dd>


<dt>num.hidden.layers</dt>
<dd><p>Number of hidden layers of the neural network (2 by
default). This number must be equal to the length of <code>num.units</code>
argument.</p></dd>


<dt>num.units</dt>
<dd><p>Vector indicating the number of neurons per hidden layer
(<code>c(200, 200)</code> by default). The length of this vector must be equal to
the <code>num.hidden.layers</code> argument.</p></dd>


<dt>activation.fun</dt>
<dd><p>Activation function (<code>'relu'</code> by default). See
the
<a href="https://tensorflow.rstudio.com/reference/keras/activation_relu.html" class="external-link">keras
documentation</a> to know available activation functions.</p></dd>


<dt>dropout.rate</dt>
<dd><p>Float between 0 and 1 indicating the fraction of
input neurons to be dropped in layer dropouts (0.25 by default). By
default, <span class="pkg">SpatialDDLS</span> implements 1 dropout layer per hidden layer.</p></dd>


<dt>loss</dt>
<dd><p>Character indicating loss function selected for model training
(<code>'kullback_leibler_divergence'</code> by default). See the
<a href="https://tensorflow.rstudio.com/reference/keras/loss-functions.html" class="external-link">keras
documentation</a> to know available loss functions.</p></dd>


<dt>metrics</dt>
<dd><p>Vector of metrics used to assess model performance during
training and evaluation (<code>c("accuracy", "mean_absolute_error",
  "categorical_accuracy")</code> by default). See the
<a href="https://tensorflow.rstudio.com/reference/keras/metric_binary_accuracy.html" class="external-link">keras
documentation</a> to know available performance metrics.</p></dd>


<dt>normalize</dt>
<dd><p>Whether to normalize data using logCPM (<code>TRUE</code> by
default). This parameter is only considered when the method used to
simulate mixed transcriptional profiles (<code>simMixedProfiles</code>
function) was <code>"AddRawCount"</code>. Otherwise, data were already
normalized.</p></dd>


<dt>scaling</dt>
<dd><p>How to scale data before training. It can be:
<code>"standardize"</code> (values are centered around the mean with a unit
standard deviation), <code>"rescale"</code> (values are shifted and rescaled so
that they end up ranging between 0 and 1) or <code>"none"</code> (no
scaling is performed). <code>"standardize"</code> by default.</p></dd>


<dt>norm.batch.layers</dt>
<dd><p>Whether to include batch normalization layers
between each hidden dense layer (<code>TRUE</code> by default).</p></dd>


<dt>custom.model</dt>
<dd><p>It allows to use a custom neural network architecture. It
must be a <code>keras.engine.sequential.Sequential</code> object in which the
number of input neurons is equal to the number of considered
features/genes, and the number of output neurons is equal to the number of
cell types considered (<code>NULL</code> by default). If provided, the arguments
related to the neural network architecture will be ignored.</p></dd>


<dt>shuffle</dt>
<dd><p>Boolean indicating whether data will be shuffled (<code>TRUE</code>
by default).</p></dd>


<dt>sc.downsampling</dt>
<dd><p>It is only used if <code>type.data.train</code> is equal to
<code>'both'</code> or <code>'single-cell'</code>. It allows to set a maximum number of
single-cell profiles of a specific cell type for training to avoid
an unbalanced representation of classes (<code>NULL</code> by default).</p></dd>


<dt>use.generator</dt>
<dd><p>Boolean indicating whether to use generators during
training and test. Generators are automatically used when <code>on.the.fly
  = TRUE</code> or HDF5 files are used, but it can be activated by the user on
demand (<code>FALSE</code> by default).</p></dd>


<dt>on.the.fly</dt>
<dd><p>Boolean indicating whether simulated data will be generated
'on the fly' during training (<code>FALSE</code> by default).</p></dd>


<dt>agg.function</dt>
<dd><p>If <code>on.the.fly == TRUE</code>, function used to build
mixed transcriptional profiles. It may be:</p><ul><li><p><code>"AddRawCount"</code> (by default): single-cell
profiles (raw counts) are added up across cells. Then, log-CPMs are
calculated.</p></li>
<li><p><code>"MeanCPM"</code>: single-cell profiles (raw counts) are
transformed into logCPM and cross-cell averages are calculated.</p></li>
<li><p><code>"AddCPM"</code>: single-cell profiles (raw counts) are transformed
into CPMs and are added up across cells. Then, log-CPMs are calculated.</p></li>
</ul></dd>


<dt>threads</dt>
<dd><p>Number of threads used during simulation of mixed
transcriptional profiles if <code>on.the.fly = TRUE</code> (1 by default).</p></dd>


<dt>view.metrics.plot</dt>
<dd><p>Boolean indicating whether to show plots of loss and
evaluation metrics during training (<code>TRUE</code> by default). <span class="pkg">keras</span>
for R allows to see model progression during training if you are working in
RStudio.</p></dd>


<dt>verbose</dt>
<dd><p>Boolean indicating whether to display model progression during
training and model architecture information (<code>TRUE</code> by default).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> object with <code>trained.model</code></p>


<p>slot containing a <code><a href="DeconvDLModel-class.html">DeconvDLModel</a></code> object. For more
information about the structure of this class, see
<code>?<a href="DeconvDLModel-class.html">DeconvDLModel</a></code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><strong>Simulation of mixed transcriptional profiles 'on the fly'</strong></p>
<p><code>trainDeconvModel</code> can avoid storing simulated mixed spot profiles by
using the <code>on.the.fly</code> argument. This functionality aims at reducing the
the <code>simMixedProfiles</code> function's memory usage: simulated profiles are
built in each batch during training/evaluation.</p>
<p><strong>Neural network architecture</strong></p>
<p>It is possible to change the model's architecture: number of hidden layers,
number of neurons for each hidden layer, dropout rate, activation function,
and loss function. For more customized models, it is possible to provide a
pre-built model through the <code>custom.model</code> argument (a
<code>keras.engine.sequential.Sequential</code> object) where it is necessary that
the number of input neurons is equal to the number of considered
features/genes, and the number of output neurons is equal to the number of
considered cell types.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="plotTrainingHistory.html">plotTrainingHistory</a></code> <code><a href="deconvSpatialDDLS.html">deconvSpatialDDLS</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>      <span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">30</span>, lambda <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">15</span>, ncol <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>      dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Cell_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RHC"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    Cell_Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"CellType"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>                       replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  rowData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    Gene_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="createSpatialDDLSobject.html">createSpatialDDLSobject</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sc.data <span class="op">=</span> <span class="va">sce</span>,</span></span>
<span class="r-in"><span>  sc.cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  sc.gene.ID.column <span class="op">=</span> <span class="st">"Gene_ID"</span>,</span></span>
<span class="r-in"><span>  sc.filt.genes.cluster <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Spatial transcriptomics data not provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Processing single-cell data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       - Filtering features:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Selected features: 15</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>          - Discarded features: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === No mitochondrial genes were found by using ^mt- as regrex</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Final number of dimensions for further analyses: 15</span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="genMixedCellProp.html">genMixedCellProp</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  cell.ID.column <span class="op">=</span> <span class="st">"Cell_ID"</span>,</span></span>
<span class="r-in"><span>  cell.type.column <span class="op">=</span> <span class="st">"Cell_Type"</span>,</span></span>
<span class="r-in"><span>  num.sim.spots <span class="op">=</span> <span class="fl">50</span>,</span></span>
<span class="r-in"><span>  train.freq.cells <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span></span>
<span class="r-in"><span>  train.freq.spots <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,</span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === The number of mixed profiles that will be generated is equal to 50</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Training set cells by type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType1: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType2: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Test set cells by type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType1: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - CellType2: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Probability matrix for training data:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Mixed spots: 34</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Cell types: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Probability matrix for test data:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Mixed spots: 16</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     - Cell types: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu"><a href="simMixedProfiles.html">simMixedProfiles</a></span><span class="op">(</span><span class="va">SDDLS</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Setting parallel environment to 1 thread(s)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Generating train mixed profiles:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Generating test mixed profiles:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-in"><span><span class="va">SDDLS</span> <span class="op">&lt;-</span> <span class="fu">trainDeconvModel</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">SDDLS</span>,</span></span>
<span class="r-in"><span>  batch.size <span class="op">=</span> <span class="fl">12</span>,</span></span>
<span class="r-in"><span>  num.epochs <span class="op">=</span> <span class="fl">5</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Training and test from stored data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Using only simulated mixed samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Using only simulated mixed samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Model: "SpatialDDLS"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Layer (type)                   Output Shape               Param #    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> =====================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dense1 (Dense)                 (None, 200)                3200       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> BatchNormalization1 (BatchNorm (None, 200)                800        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Activation1 (Activation)       (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dropout1 (Dropout)             (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dense2 (Dense)                 (None, 200)                40200      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> BatchNormalization2 (BatchNorm (None, 200)                800        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Activation2 (Activation)       (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dropout2 (Dropout)             (None, 200)                0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dense3 (Dense)                 (None, 2)                  402        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> BatchNormalization3 (BatchNorm (None, 2)                  8          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ActivationSoftmax (Activation) (None, 2)                  0          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> =====================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Total params: 45,410</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Trainable params: 44,606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-trainable params: 804</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _____________________________________________________________________</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Training DNN with 34 samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Evaluating DNN in test data (16 samples)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - loss: NaN</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - accuracy: 0.5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - mean_absolute_error: NaN</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    - categorical_accuracy: 0.5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> === Generating prediction results using test data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span>  </span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

